### ISSUE TITLE
Binary cross entropy on MNIST data

### ISSUE THREAD

**Issue by samadejacobs** [LLNL/lbann#186]
I have seen binary cross entropy loss function used in a couple of public blogs on MNIST dataset as a substitute for MSE (l[ink1](https://blog.keras.io/building-autoencoders-in-keras.html))([link2)](https://github.com/stitchfix/fauxtograph/issues/15). LBANN currently doesn't support this: it segfaults with error message: /usr/workspace/wsa/jacobs32/timmoon10-lbann-layerrf/src/objective_functions/loss_functions/binary_cross_entropy.cpp 56 :: binary cross entropy loss function requires binary ground truth.
We may want to revisit the formula for computing binary cross entropy with non-binary ground truth
